### ISSUE TITLE
Cannot return HTTP header using "Grpc-Metadata-" prefix

### ISSUE THREAD

**Issue by joelurraco** [grpc-ecosystem/grpc-gateway#782]
I have been using the following line of code to return HTTP headers in the server response:

```
import (
    "google.golang.org/grpc"
    "google.golang.org/grpc/metadata"
)

grpc.SetHeader(ctx, metadata.Pairs("Grpc-Metadata-Some-Key", someValue))
```

This was working OK using the `runtime.DefaultHeaderMatcher` when starting the server, but after updating this no longer works.

I went over the `google.golang.org/grpc/metadata` package and it seems that every key set in the metadata is transformed to lower case, but the `runtime.DefaultHeaderMatcher` function compares the transformed metadata key with the prefix "Grpc-Metadata-", so it never matches and the header is never returned.

I think [this](https://github.com/grpc-ecosystem/grpc-gateway/commit/8976e602c518589cb6d40aca52a7dd8aef83706a#diff-85c4fe88fcf7047845e0e64e2f4a0722L54) broke it.

Am I doing anything wrong? How should I set the metadata so it is returned as an HTTP header?

**Comment by johanbrandhorst** [comment]
Hi @joelurraco, this does indeed sound problematic. The fix seems simple enough, do you think you could contribute a fix and a new test case?

**Comment by johanbrandhorst** [comment]
Fixed with #784
